flowey/flowey_cli/src/pipeline_resolver/ado_yaml.rs

```diff
@@ -644,41 +652,50 @@ EOF
         parameters: if !parameters.is_empty() {
             Some(
                 parameters
+                    .clone()
                     .into_iter()
-                    .enumerate()
-                    .map(|(idx, param)| match param {
-                        flowey_core::pipeline::internal::Parameter::Bool {
-                            description,
-                            default,
-                        } => schema_ado_yaml::Parameter {
-                            name: format!("param{idx}"),
-                            display_name: description,
-                            ty: schema_ado_yaml::ParameterType::Boolean { default },
-                        },
-                        flowey_core::pipeline::internal::Parameter::String {
-                            description,
-                            default,
-                            possible_values,
-                        } => schema_ado_yaml::Parameter {
-                            name: format!("param{idx}"),
-                            display_name: description,
-                            ty: schema_ado_yaml::ParameterType::String {
+                    .map(|param| {
+                        let name = param.name();
+                        match param {
+                            flowey_core::pipeline::internal::Parameter::Bool {
+                                name: _,
```

Comment:
this is a bit weird to see, where you need to ignore the name field, in favor of the computed name returned by `.name()`?

instead of doing this, lets just lift the naming logic into the `new_param_XXX` methods directly, so that this code can simply read the `name` field itself, without going through a helper.

should be easy to make a free-standing `fn compute_name(name: &str, kind: ParamKind) -> String` helper method to reuse across all `new_param_XXX` calls.